Source/JavaScriptCore/heap/MarkedBlock.cpp

```diff
@@ -502,6 +506,67 @@ void MarkedBlock::Handle::sweep(FreeList* freeList)
     specializedSweep<false, IsEmpty, SweepOnly, BlockHasNoDestructors, DontScribble, HasNewlyAllocated, MarksStale>(freeList, emptyMode, sweepMode, BlockHasNoDestructors, scribbleMode, newlyAllocatedMode, marksMode, [] (VM&, JSCell*) { });
 }
 
+NO_RETURN_DUE_TO_CRASH NEVER_INLINE void MarkedBlock::dumpInfoIfHandleIsNotValid(AbstractLocker&, JSCell* cell)
+{
+    size_t contiguousZeroCount = 0;
+    {
+        char* mem = WTF::bitwise_cast<char*>(&header());
+        for (; contiguousZeroCount < sizeof(MarkedBlock::blockSize); ++contiguousZeroCount) {
+            if (*mem)
+                break;
+            ++mem;
+        }
+    }
+
+    bool isValidBlockVM = false;
+    bool isBlockInVM = false;
+    bool isBlockHandleInVM = false;
+    VM* blockVM = header().m_vm;
+    {
+        isValidBlockVM = blockVM ? VMInspector::isValidVM(blockVM) : false;
+        if (isValidBlockVM) {
```

Comment:
We know the pointer to handle in the header is corrupted (or we are already in some region that isn't actually a block), so it seems like there's a good chance m_vm field is also corrupted, and it may be corrupted with non-null values. So is there any value to having this 'then' path rather than just using the 'else' path always?